lib/mix/tasks/phx.gen.html.ex

```diff
@@ -2,54 +2,66 @@ defmodule Mix.Tasks.Phx.Gen.Html do
-    * default CRUD templates in `lib/app_web/controllers/user_html`
-
-  ## The context app
-
-  A migration file for the repository and test files for the context and
-  controller features will also be generated.
-
-  The location of the web files (controllers, HTML views, templates, etc) in an
-  umbrella application will vary based on the `:context_app` config located
-  in your applications `:generators` configuration. When set, the Phoenix
-  generators will generate web files directly in your lib and test folders
-  since the application is assumed to be isolated to web specific functionality.
-  If `:context_app` is not set, the generators will place web related lib
-  and test files in a `web/` directory since the application is assumed
-  to be handling both web and domain specific functionality.
-  Example configuration:
-
-      config :my_app_web, :generators, context_app: :my_app
+  > over distinct contexts (e.g., `Accounts.User` and `Payments.User`).
+
+  This generator adds the following files to `lib/`:
+
+    * a controller in `lib/my_app_web/controllers/user_controller.ex`
+    * default CRUD HTML templates in `lib/my_app_web/controllers/user_html`
+    * an HTML view collocated with the controller in `lib/my_app_web/controllers/user_html.ex`
+    * a schema in `lib/my_app/accounts/user.ex`, with an `users` table
+    * a context module in `lib/my_app/accounts.ex` for the accounts API
+  
+  Additionally, this generator creates the following files:
+    
+    * a migration for the schema in `priv/repo/migrations`
+    * a controller test module in `test/my_app/controllers/user_controller_test.exs`
+    * a context test module in `test/my_app/accounts_test.exs`
+    * a context test helper module in `test/support/fixtures/accounts_fixtures.ex` 
+  
+  If the context already exists, this generator injects functions for the given resource into
+  the context, context test, and context test helper modules.
+
+  ## Umbrella app configuration
+
+  `my_app_umbrella` assumes that `my_app_web` will handle both web and domain specific functionality. 
+  You can change the location of your controllers — with collocated views — and template files by
+  changing the value of the `:context_app` configuration option.
+  
+  Phoenix generators will generate web files directly in your lib and test folders since 
+  the application is assumed to be isolated to web specific functionality.
+  
+  Example configuration in `my_app_umbrella/config/config.exs`:
+
+    config :my_app_web,
+      ecto_repos: [Stuff.Repo],
+      generators: [context_app: :my_app]
 
   Alternatively, the `--context-app` option may be supplied to the generator:
 
-      mix phx.gen.html Sales User users --context-app warehouse
+    mix phx.gen.html Sales User users --context-app my_app
+  
+  If you delete the `:context_app` configuration option, Phoenix will automatically put generated web files in
+  `my_app_umbrella/apps/my_app_web_web`.
```

Comment:
Is the `_web_web` a typo or what happens? If so, it is a bug.

Reply:
That is what happens right now. It appends a _web to whatever is in the congfig file

Reply:
I see. It seems what is happening here is the following: umbrellas assume you will have two applications, `my_app` and a `my_app_web`, where `my_app` is the context app. So perhaps we should document this assumption explicitly instead?

Reply:
In any case, besides documenting the assumptions, it seems the my_app_web_web has to be addressed. Please file a bug report. For simplicity, we can actually assume the web app to be whatever app the generator is running on.